tensorrt_llm/_torch/pyexecutor/py_executor.py

```
@@ -3363,12 +3374,131 @@ def _update_requests(self,
             logger.error(f"Encountered an error in sampling: {error_msg}")
             self._handle_errors(error_msg)
 
+    # Patterns that corrupt the CUDA context beyond recovery.
+    # Matched case-insensitively against the error message.
+    _IMMEDIATE_FATAL_PATTERNS: list[str] = [
+        "cudaerrorillegaladd",
+        "cudaerrorlaunchfailure",
+        "device-side assert",
+        "unrecoverable",
+    ]
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)